### ISSUE TITLE
[CHILD] Research for object model

### ISSUE THREAD

**Issue by reomelo** [SEAME-pt/HotWheels-Lane_Detection#71]
`Description:` Small research on methods and tools to use in the development of the model. What dataset? What libraries (tensorflow, pytorch, ect)?

</br>

`Definition of done:` This issue is considered as done when the previous questions have been answered and we have a defined course of action

**Comment by reomelo** [comment]
After some research I concluded that YOLO is the best object detection model at the moment. Before explaining why here are some key concepts to better understand the topic:

### Concepts
**Mean Average Prediction (mAP)** - metric used to evaluate object detection models based on sub metrics such as Confusion Matrix, Intersection over Union (IoU), Recall, and Precision. ***YOLO has a mAP of 57.9% on the COCO dataset, higher than any other model.***

COCO (Common Objects in Context) - is a large scale dataset created to train and evaluate object detection algorithms. It has more detail per image than any other dataset and supports many tasks such as Detection, Segmentation, Keypoints and Captioning. It uses the PyTorch framework.

COCO's content:
- 330k images
- 200k labeled images
- 1.5M object instances
- 80 object types (car, dog, person, etc)
- segmentaion masks for each object

COCO's division:
- Train2017 (118k images)
- Val2017 (5k images)
- Test2017 (unlabeled images)

### YOLO model
While other models propose potential locations and then classify them, YOLO performs both tasks at the same time and achieves greater speed and accurary values. ***76.5% accuracy in traffic management systems (top 1) and 93.3% accuracy in vehicle detection (top 5).***

### Structure
Our project structure should be similar to this:
- Library: YOLOv8
- Dataset: COCO
- Labeling Tool: Roboflow (only needed if fine-tuning with custom data)
- Training: Use built-in Ultralytics trainer

Source -> https://www.v7labs.com/blog/mean-average-precision
